### PR TITLE
fix: shaka.polyfill.MediaSource.rejectCodec_() incorrectly handles mime-type

### PR DESCRIPTION
**Pull request by gmcgarry**
For example:

        shaka.polyfill.MediaSource.rejectCodec_('dvh1')

Useful to disable matching against broken codecs, even though it's supposed to be a private method.

This function is used in some conditional code, but I cannot see how it could be working correctly.

Notes:

1) shaka.util.MimeUtils.getCodecBase() does not take a mime-type as input.  It takes a codecs parameter.
2) add call to shaka.util.MimeUtils.getCodecs() to get the codecs parameter first                                                                                                           

Added some tests for this function, but difficult to run the test in isolation.

### CODE REVIEW

**Comment by avelad** [comment]
@gmcgarry can you fix the lint errors? Thanks!

File: lib/polyfill/mediasource.js [hunk 1]
```diff
@@ -214,7 +215,8 @@ shaka.polyfill.MediaSource = class {
           ManagedMediaSource.isTypeSupported.bind(ManagedMediaSource);
 
       window.ManagedMediaSource.isTypeSupported = (mimeType) => {
-        const actualCodec = shaka.util.MimeUtils.getCodecBase(mimeType);
+        const codecs = shaka.util.MimeUtils.getCodecs(mimeType);
```
Comment (joeyparrish) [inline lib/polyfill/mediasource.js:218]: Code change looks good to me.  Only the tests need improvement AFAICT.  Thank you for contributing!

Comment (joeyparrish) [reply lib/polyfill/mediasource.js:218]: Also, please take a look at the linter errors.  You can run `python3 build/check.py` to see them locally before your next push.

File: test/polyfill/media_source_unit.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+describe('MediaSource', () => {
+  const originalMediaSource = window.MediaSource;
+
+  afterEach(() => {
+    window.MediaSource = originalMediaSource;
+  });
```
[the remark this answers is not in the document]
Comment (joeyparrish) [reply test/polyfill/media_source_unit.js:12]: This is still broken.  Also, you have some failing tests.

**Pull request closed without merging**